scss/components/_off-canvas.scss

```diff
@@ -266,16 +340,35 @@ $content: $maincontent-class
   transform: none;
   z-index: $zindex;
   transition: none;
+  visibility: visible;
 
   @if not $offcanvas-fixed-reveal {
     position: absolute;
   }
 
+  @at-root .#{$content}.has-reveal-#{$position} {
+    margin-#{$position}: $offcanvas-size;
+  }
+
+  // backwards compatibility (prior to v6.4)
   & ~ .#{$content} {
     margin-#{$position}: $offcanvas-size;
   }
 }
 
+/// Overrides the off-canvas styles
+@mixin in-canvas() {
```

Comment:
I've used !important here due to the CSS specificity of the things I'm overriding.
Otherwise every property I've set important wouldn't work with only one simple in-canvas selector.
I would have to fight against the specificity of the following selectors:

- transform => .off-canvas-content .off-canvas.position-[DIRECTION]
- visibility => .off-canvas.is-closed
- height => inline style set by JS (fixed position emulation)
- box-shadow => .off-canvas-content .position-[DIRECTION].is-transition-push
- position => .off-canvas-content .off-canvas

I agree !important is not cool but I think in this case it's better than including in-canvas into all the long selectors (inline style is even impossible without important).
It's worth to mention I only used !important where necessary. Some properties of my in-canvas feature are working without it (background, width, overflow, transition) and just use 'inherit'.